payment/src/main/java/org/killbill/billing/payment/core/sm/PaymentAutomatonRunner.java

```diff
@@ -128,8 +129,10 @@ public PaymentStateContext buildPaymentStateContext(final boolean isApiPayment,
                                                         final Iterable<PluginProperty> properties,
                                                         final CallContext callContext,
                                                         final InternalCallContext internalCallContext) throws PaymentApiException {
+
         // Retrieve the payment id from the payment external key if needed
         final UUID effectivePaymentId = paymentId != null ? paymentId : retrievePaymentId(paymentExternalKey, paymentTransactionExternalKey, internalCallContext);
+        verifyPaymentMethodExistsForRefund(effectivePaymentId, transactionType, internalCallContext);
```

Comment:
I suspect the reason for `getPaymentMethodIncludedDeleted` is that when following payment operations we want(ed) to make sure we go through the same plugin -> gateway. By retrieving the original payment method (even though it could now be deleted), we are able to extract the `pluginName` and route the following operation towards same plugin -> payment gateway. 

In situation where we have payment control plugin this is probably not mandatory, but in other situations this *may* be needed.